cmds/webboot/webboot.go

```diff
@@ -122,54 +172,61 @@ func (d *DirOption) exec(uiEvents <-chan ui.Event) (menu.Entry, error) {
 	return menu.DisplayMenu("Distros", "Choose an option", entries, uiEvents)
 }
 
-func main() {
-	flag.Parse()
-	if *v {
-		verbose = log.Printf
-	}
-
+func getMainMenu() menu.Entry {
 	entries := []menu.Entry{
+		&UseCacheOption{},
 		&DownloadOption{},
 	}
-
-	if *dir == "" {
-		mp, err := getCachedDirectory()
-		if err == nil {
-			entries = append(entries, &DirOption{
-				label: "Use Cached ISO",
-				path:  filepath.Join(mp.Path, "Image"),
-			})
-		}
-	} else {
-		entries = append(entries, &DirOption{
-			label: "Use Cached ISO",
-			path:  *dir,
-		})
-	}
-
-	entry, err := menu.DisplayMenu("Webboot", "Choose an ISO:", entries, ui.PollEvents())
+	entry, err := menu.DisplayMenu("Webboot", "Choose an option:", entries, ui.PollEvents())
 	if err != nil {
 		log.Fatal(err)
 	}
+	return entry
+}
+
+func main() {
+	flag.Parse()
+	if *v {
+		verbose = log.Printf
+	}
 
+	entry := getMainMenu()
+	var err error = nil
 	// check the chosen entry of each level
 	// and call it's exec() to get the next level's chosen entry.
 	// repeat this process until there is no next level
 	for entry != nil {
 		switch entry.(type) {
+		case *UseCacheOption:
```

Comment:
Actually when I look at it I cannot remember the reason I have this separate struct at that time i wrote, But I remember there is a reason. Anyway, I'll change it and see if there is a problem.

Reply:
I know what the reason is now. It is because in my image, if the cache dir is not exist, I will also show a option of "use cached ISO". and if it is hit, the menu only contains back option. And DirOption can not doing this because it cannot represent a path do not exist so I need UseCacheOption.  But when I think further, just do not show the "use cached ISO" option at all when i can not find the cache dir would be simpler. I'll change it

Reply:
Sounds good!